### PR TITLE
feat: entity linking

### PR DESCRIPTION
**Pull request by skonigs**

### CODE REVIEW

File: cognitive/src/main/scala/com/microsoft/ml/spark/cognitive/CognitiveServiceBase.scala [hunk 1]
```diff
@@ -139,7 +139,18 @@ trait HasSubscriptionKey extends HasServiceParams {
...
+    this, name = "options", "options for endpoint")
+
+  def getOptions: Option[TextAnalyticsRequestOptionsV4] = get(options)
+
+  def setOptions(v: TextAnalyticsRequestOptionsV4): this.type = set(options, v)
+
+}
+
+
+    object URLEncodingUtils {
```
Comment (suhas92) [inline cognitive/src/main/scala/com/microsoft/ml/spark/cognitive/CognitiveServiceBase.scala:153]: fix indent

File: cognitive/src/test/scala/com/microsoft/ml/spark/cognitive/split1/TextAnalyticsSDKSuite.scala [hunk 2]
```diff
@@ -28,11 +28,12 @@ class DetectedLanguageSuiteV4 extends TestBase with DataFrameEquality with TextK
   def getDetector: LanguageDetectionV4 = new LanguageDetectionV4()
     .setSubscriptionKey(textKey)
     .setLocation("eastus")
+    .setOptions(TextAnalyticsRequestOptionsV4("", true, false))
```
Comment (suhas92) [inline cognitive/src/test/scala/com/microsoft/ml/spark/cognitive/split1/TextAnalyticsSDKSuite.scala:31]: what are the 3 params here?

File: cognitive/src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala [hunk 3]
```diff
@@ -229,3 +229,23 @@ class HealthcareV4(override val uid: String) extends TextAnalyticsSDKBase[Health
     toResponse(poller.getFinalResult.asScala.flatMap(_.asScala))
   }
 }
+
+object EntityLinkingV4 extends ComplexParamsReadable[EntityLinkingV4]
+
+class EntityLinkingV4(override val uid: String) extends TextAnalyticsSDKBase[LinkedEntityCollectionV4]() {
+  logClass()
+
+  def this() = this(Identifiable.randomUID("PIIV4"))
```
Comment (suhas92) [inline cognitive/src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala:238]: should this be something else than PIIV4?

File: cognitive/src/test/scala/com/microsoft/ml/spark/cognitive/split1/TextAnalyticsSDKSuite.scala [hunk 4]
```diff
@@ -557,4 +560,84 @@ class HealthcareSuiteV4 extends TestBase with DataFrameEquality with TextKey {
...
+      .collect()
+
+    val firstRow = replies(0)
+    assert(replies(0).schema(0).name == "output")
+    val fromRow = LinkedEntityResponseV4.makeFromRowConverter
+    val resFirstRow = fromRow(firstRow.getAs[GenericRowWithSchema]("output"))
+    val linkedEntities = resFirstRow.result.head.get.entities
+    assert(linkedEntities.nonEmpty)
+
+    println("Entities:")
```
Comment (HanLiMS) [inline cognitive/src/test/scala/com/microsoft/ml/spark/cognitive/split1/TextAnalyticsSDKSuite.scala:605]: This is only for printing. Might want to add some assertions here?

File: cognitive/src/test/scala/com/microsoft/ml/spark/cognitive/split1/TextAnalyticsSDKSuite.scala (hunk 2 above)
Comment (skonigs) [reply cognitive/src/test/scala/com/microsoft/ml/spark/cognitive/split1/TextAnalyticsSDKSuite.scala:31]: model version , stats, disable service log

**Comment by suhas92** [comment]
Looks good 💯

**Pull request merged**